core/src/main/java/org/apache/calcite/sql/dialect/StarRocksSqlDialect.java

```diff
@@ -128,6 +132,52 @@ public StarRocksSqlDialect(Context context) {
               timeUnitNode.getParserPosition());
       SqlFloorFunction.unparseDatetimeFunction(writer, newCall, "DATE_TRUNC", false);
       break;
+    case IS_TRUE:
```

Comment:
this won't work if the expression has side effects, e.g., the function is RAND().

Reply:
Good catch. I overlooked this non-deterministic expression. If this expression is non-deterministic, I will throw an exception. Or do you have any other more general approaches for handling this situation?

Reply:
The only solution I can think of is more complicated: you need to add a new project saving the value of the expression, and then do the expansion for that value.

Reply:
Ok, I understand your approach. For this PR, I will throw an exception for non-deterministic expressions.

Reply:
Perhaps we'll consider your suggestion later.